command/agent/auth/auth.go

```diff
@@ -151,6 +151,38 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) error {
 			clientToUse = ah.client
 		}
 
+		var secret *api.Secret = new(api.Secret)
+		if first && ah.token != "" {
+			ah.logger.Debug("using preloaded token")
+
+			first = false
+			ah.logger.Debug("lookup-self with preloaded token")
+			clientToUse.SetToken(ah.token)
+
+			secret, err = clientToUse.Logical().Read("auth/token/lookup-self")
+			if err != nil {
+				ah.logger.Error("could not look up token", "err", err, "backoff", backoff.Seconds())
+				backoffOrQuit(ctx, backoff)
```

Comment:
I appreciate it's an inherited pattern, but is it worth considering putting `backoffOrQuit()` calls at the beginning of the loop behind a `if !first` check? It would simplify `continue` blocks, and reduce both duplication and the potential for bugs where future developers forget to call `backoffOrQuit`.

Reply:
Honestly this code doesn't change much and is likely going to be refactored when agent code is decoupled from Vault, so I think we'll defer this for now.